amp.php

```diff
@@ -50,19 +51,55 @@ function amp_deactivate() {
 	flush_rewrite_rules();
 }
 
-AMP_Post_Type_Support::init();
-
-add_action( 'init', 'amp_init' );
-function amp_init() {
+/**
+ * Set up AMP.
+ *
+ * This function must be invoked through the 'after_setup_theme' action to allow
+ * the AMP setting to declare the post types support earlier than plugins/theme.
+ *
+ * @since 0.6
+ */
+function amp_after_setup_theme() {
 	if ( false === apply_filters( 'amp_is_enabled', true ) ) {
 		return;
 	}
 
+	if ( ! defined( 'AMP_QUERY_VAR' ) ) {
+		/**
+		 * Filter the AMP query variable.
+		 *
+		 * @since 0.3.2
+		 * @param string $query_var The AMP query variable.
+		 */
+		define( 'AMP_QUERY_VAR', apply_filters( 'amp_query_var', 'amp' ) );
+	}
+
+	add_action( 'init', 'amp_init' );
+	add_action( 'admin_init', 'AMP_Options_Manager::register_settings' );
+	add_filter( 'amp_post_template_analytics', 'amp_add_custom_analytics' );
+	add_action( 'wp_loaded', 'amp_post_meta_box' );
+	add_action( 'wp_loaded', 'amp_add_options_menu' );
+	AMP_Post_Type_Support::add_post_type_support();
+}
+add_action( 'after_setup_theme', 'amp_after_setup_theme', 5 );
+
+/**
+ * Init AMP.
+ *
+ * @since 0.1
+ */
+function amp_init() {
+
+	/**
+	 * Triggers on init when AMP plugin is active.
+	 *
+	 * @since 0.3
+	 */
 	do_action( 'amp_init' );
 
 	load_plugin_textdomain( 'amp', false, plugin_basename( AMP__DIR__ ) . '/languages' );
 
-	amp_define_query_var();
+	amp_after_setup_theme();
```

Comment:
Any specific reason to call `amp_after_setup_theme()` here since the `amp_init()` is invoked in a `init` action which is declared in the `amp_after_setup_theme()`?

Reply:
Good catch. Yeah, this should have already been called so it should be removed entirely.